hazelcast/src/main/java/com/hazelcast/internal/cluster/impl/MembershipManager.java

```diff
@@ -367,6 +367,7 @@ private MemberImpl createNewMemberImplIfChanged(MemberInfo newMemberInfo, Member
                 member = clusterService.promoteAndGetLocalMember();
             } else {
                 member = createMember(newMemberInfo, member.getAttributes());
+                node.partitionService.memberAdded(member);
```

Comment:
Shouldn't this be out of `else` block? What if this is on local member?

Reply:
On the local member the  `node.partitionService.memberAdded(member);` is doing update only when the node is not local. For local case, it issues a special task only when the node is a master. But the master is notified about promotion without this change. This is why I decided that it is enough to put it in the `else`.

Reply:
Also, the problem is when the member is not aware of **other**  members being promoted. That case is covered in the `else` branch

Reply:
I think issue is a non-master member is not aware of any member being promoted, including itself. So maybe we should modify `node.partitionService.memberAdded(member)` to update `memberGroupSize even with local member.

See this modified test: